core-bundle/src/EventListener/DataContainer/DefaultOperationsListener.php

```diff
@@ -0,0 +1,222 @@
+            ];
+        }
+
+        $operations['delete'] = [
+            'href' => 'act=delete',
+            'icon' => 'delete.svg',
+            'attributes' => 'onclick="if(!confirm(\''.($GLOBALS['TL_LANG']['MSC']['deleteConfirm'] ?? null).'\'))return false;Backend.getScrollOffset()"',
+            'button_callback' => $this->isGrantedCallback(DeleteAction::class, $table),
+        ];
+
+        if (null !== ($toggleField = $this->getToggleField($table))) {
+            $operations['toggle'] = [
+                'href' => 'act=toggle&amp;field='.$toggleField,
+                'icon' => 'visible.svg',
+                'button_callback' => $this->isGrantedCallback(UpdateAction::class, $table),
+            ];
+        }
+
+        return $operations + [
+            'show' => [
+                'href' => 'act=show',
+                'icon' => 'show.svg',
+            ],
+        ];
+    }
+
+    private function isGrantedCallback(string $actionClass, string $table): \Closure
+    {
+        return function (DataContainerOperation $operation) use ($actionClass, $table): void {
+            if (!$this->isGranted($actionClass, $table, $operation)) {
+                $this->disableOperation($operation);
+            }
+        };
+    }
+
+    private function copyChildsCallback(string $table): \Closure
+    {
+        return function (DataContainerOperation $operation) use ($table): void {
+            if (!$this->isGranted(CreateAction::class, $table, $operation)) {
+                $this->disableOperation($operation);
+
+                return;
+            }
+
+            $childCount = $this->connection->fetchOne(
+                "SELECT COUNT(*) FROM $table WHERE pid=?",
+                [(string) $operation->getRecord()['id']]
+            );
+
+            if ($childCount < 1) {
+                $this->disableOperation($operation);
+            }
+        };
+    }
+
+    /**
+     * Finds the one and only toggle field in a DCA. Returns null if multiple fields can be toggled.
+     */
+    private function getToggleField(string $table): string|null
+    {
+        $field = null;
+
+        foreach ($GLOBALS['TL_DCA'][$table]['fields'] as $name => $config) {
+            if (!($config['toggle'] ?? false) && !($config['reverseToggle'] ?? false)) {
+                continue;
+            }
+
+            // more than one toggle field exists
+            if (null !== $field) {
+                return null;
+            }
+
+            $field = $name;
+        }
+
+        return $field;
+    }
+
+    private function isGranted(string $actionClass, string $table, DataContainerOperation $operation): bool
+    {
+        $subject = match ($actionClass) {
+            CreateAction::class => new CreateAction($table),
+            UpdateAction::class => new UpdateAction($table, $operation->getRecord()),
+            DeleteAction::class => new DeleteAction($table, $operation->getRecord())
```

Comment:
```suggestion
            DeleteAction::class => new DeleteAction($table, $operation->getRecord()),
            default => throw new \InvalidArgumentException(sprinf('Invalid action class "%s".', $actionClass)),
```

Reply:
I don't think that's necessary, we're the only ones calling that method and we know our options